ayon_server/addons/addon.py

```diff
@@ -678,4 +682,71 @@ async def execute_action(
         executor: ActionExecutor,
     ) -> ExecuteResponseModel:
         """Execute an action provided by the addon"""
-        raise ValueError(f"Unknown action: {executor.identifier}")
+        raise BadRequestException(f"Unknown action: {executor.identifier}")
+
+    async def create_config_hash(
+        self,
+        identifier: str,
+        context: ActionContext,
+        user: UserEntity,
+        variant: str,
+    ) -> str:
+        """Create a hash for action config store"""
+        hash_content = [
+            user.name,
+            identifier,
+            context.project_name,
+        ]
+        logger.trace(f"Creating config hash from {hash_content}")
+        return hash_data(hash_content)
+
+    async def set_action_config(
+        self,
+        identifier: str,
+        context: ActionContext,
+        user: UserEntity,
+        variant: str,
+        config: dict[str, Any],
+    ) -> None:
+        """Set action config provided by the addon"""
+        hash = await self.create_config_hash(
```

Comment:
`hash` is built-in function.
```suggestion
        config_hash = await self.create_config_hash(
```